[Tohtoroo] iP

Tohtoroo opened the pull request:
> "Everybody, try laughing. Then whatever scares you will go away!” - Tatsuo Kusakabe (My Neighbour Totoro)

Tohtoro frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download from [here](https://github.com/Tohtoroo/ip/releases/tag/A-Jar)
2. Double-click it
3. Add your tasks
4. Let it manage tasks for you 💯 

And it is ***FREE***!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Tohtoroo committed:
Add Level 1 - Greet, Echo, Exit

Tohtoroo committed:
Add Level 2 - Add, List

Tohtoroo committed:
Add Level 3 - Mark as Done

Tohtoroo committed:
Added Abstraction + Level 4 - ToDos, Events, Deadlines

Tohtoroo committed:
Added Level 5 - Handle Errors

Tohtoroo committed:
Add Level 6 - Delete

Tohtoroo committed:
Add A-TextUITesting

muhammadreyaaz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,156 @@
+    //I'm going to start abstracting functions otherwise I'll puke blood
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
```
LGTM 

Some cosmetic change for custom duke string required.

Tohtoroo committed:
Add packages

Tohtoroo committed:
Add Level-8 Dates and Time

zhacatomn commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,109 @@
+            if(i == 0 && s.charAt(i) == '-') {
+                if(s.length() == 1) {
+                    return false;
+                } else {
```
```suggestion
               
```
You can just omit the `else` here.

zhacatomn reviewed (commented):
Do remember to add javadocs when you can. All around, looks good! 
_(PS: I love the Totoro)_

minosx31 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+    protected String by;
+
+
```
Perhaps this line break was accidental? Please ignore if it was intentional.

minosx31 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,48 @@
+
+    public abstract String[] encode();
+
+
+
```
Probably also accidental line breaks here but please if ignore if it is intentional.

minosx31 commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,107 @@
+                }
+                String description;
+                switch (type) {
+                    case "T":
+                        if (encodeSplit.size() != 4) {
+                            throw new DukeException("Error with Todo load");
+                        }
+                        description = encodeSplit.get(3);
+                        tl.add(new Todo(description, isDone));
+                        break;
+                    case "D":
+                        if (encodeSplit.size() != 5) {
+                            throw new DukeException("Error with Deadline load");
+                        }
+                        description = encodeSplit.get(3);
+                        String by = encodeSplit.get(4);
+                        tl.add(new Deadline(description, by, isDone));
+                    case "E":
+                        if (encodeSplit.size() != 6) {
+                            throw new DukeException("Error with Deadline load");
+                        }
+                        description = encodeSplit.get(3);
+                        String from = encodeSplit.get(4);
+                        String to = encodeSplit.get(5);
+                        tl.add(new Event(description, from, to, isDone));
+                    default:
```
A minor issue here is that `case` and `switch` statements should not be indented (Java coding standard) so maybe you can fix this alongside other places where `switch` is used.

minosx31 reviewed (commented):
Besides a few minor style issues, your code is mostly well organised and neat! 
Great job! 👍🏼

Tohtoroo committed:
Add Level-7 Save

Tohtoroo committed:
Merge branch 'branch-Level-8'

Tohtoroo committed:
Add A-MoreOOP

Tohtoroo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Tohtoroo committed:
Add A-Gradle

Tohtoroo committed:
Add A-JUnit

Tohtoroo committed:
Change file path for storage

Tohtoroo committed:
Add A-JavaDoc

Tohtoroo committed:
Add checkstyle configs

Tohtoroo committed:
Add A-CodingStandard

Tohtoroo committed:
Add Level-9 Find

Tohtoroo committed:
Resolve merge conflicts

Tohtoroo committed:
Resolve merge conflict Level-9

Tohtoroo committed:
Edit JavaDoc and CodingStandard

Tohtoroo committed:
GUI bug

Tohtoroo committed:
fixing GUI

Tohtoroo committed:
Add Level-10

Tohtoroo committed:
Add A-CheckStyle

Tohtoroo committed:
Merge branch 'branch-Level-10'

Tohtoroo committed:
Add Assertions to the Ui, MainWindow and Storage

The different classes have underlying conditions for the code to run.

Having assertions make notifies the developers when such conditions are false.

Therefore, I add assertions to the Ui, MainWindow and Storage class, to check for whether the input command is null, whether the GUI have pictures and when the storage does not load from a txt file in the correct template.

Tohtoroo committed:
Add and reformat JavaDoc comments

Current situation: JavaDoc comments are missing or poorly formatted.

Why it needs to change: JavaDoc comments provide crucial information to developers using the code and help maintain code quality.

What is being done about it: Add and reformat JavaDoc comments to improve code readability and provide more information to other developers.

Why it is done that way: Consistent and well-formatted JavaDoc comments make it easier for others to understand the code and can help reduce the number of bugs.

Other relevant info: A thorough review of the code has been conducted to ensure that all necessary information has been included in the comments.

Tohtoroo committed:
Improve code quality by shortening methods Load and Pase
Current situation: The methods Load and Parse are lengthy and complex.

Why it needs to change: Long and complex methods can make the code harder to understand and maintain, leading to increased risk of bugs.

What is being done about it: Improve code quality by shortening the methods Load and Parse. This is done by abstracting the exception handling to another method.

Why it is done that way: Shortening the methods makes the code easier to understand and reduces the risk of bugs. This makes maintenance and future updates easier and more efficient.

Other relevant info: Refactoring has been done to ensure that the functionality of the methods remains unchanged while improving readability and maintainability. Code comments have also been added to provide additional context and explanation for the changes.

Tohtoroo committed:
Merge pull request #3 from Tohtoroo/branch-A-Assertions

Add Assertions to the Ui, MainWindow and Storage

Tohtoroo committed:
Revert "Add Assertions to the Ui, MainWindow and Storage"

Tohtoroo committed:
Merge pull request #4 from Tohtoroo/revert-3-branch-A-Assertions

Revert "Add Assertions to the Ui, MainWindow and Storage"

Tohtoroo committed:
Revert "Revert "Add Assertions to the Ui, MainWindow and Storage""

Tohtoroo committed:
Merge pull request #5 from Tohtoroo/revert-4-revert-3-branch-A-Assertions

Revert "Revert "Add Assertions to the Ui, MainWindow and Storage""

Tohtoroo committed:
Merge branch 'master' into branch-A-CodeQuality

Tohtoroo committed:
Merge pull request #2 from Tohtoroo/branch-A-CodeQuality

Branch a code quality

Tohtoroo committed:
Debug creation of deadline and event tasks and add B-Snooze

Tohtoroo committed:
Add User Guide and debug truncation of text in GUI
